spec/features/admin/products_spec.rb

```diff
@@ -114,6 +114,71 @@
       expect(page).to have_content "Unit value can't be blank"
     end
   end
+  
+  describe "deleting", js: true do
+    let!(:product1) { create(:simple_product, name: 'a product to keep', supplier: @supplier) }
+
+    context 'a simple product' do
+      let!(:product2) { create(:simple_product, name: 'a product to delete', supplier: @supplier) }
+
+      before do
+        login_as_admin_and_visit spree.admin_products_path
+
+        within "tr#p_#{product2.id}" do
```

Comment:
I know we're using it in all feature specs but it'd be less redundant and hence robust to use `#p_#{product.id}` as selector. `#` selects id attributes and these are meant to be unique in HTML. As a result, we rely a bit less on the actual markup which will make these tests less error prone.

This is something I'd like to enforce. I've had a good experiencie doing this in other projects.

Reply:
thoughts @andrewpbrett @Matt-Yorkley @jibees ?

Reply:
The difference is just that you're removing the `tr` in case this is one day no longer a table? That makes sense to me, yeah.

Reply:
I totally agree.

Reply:
exactly, feature specs don't care whether or not this is a table, just if it works.

Reply:
Removed the `tr` tag :+1: